source/Handlebars/ValueProviders/ObjectIteratorValues.cs

```diff
@@ -7,26 +7,16 @@ namespace HandlebarsDotNet.ValueProviders
     public readonly ref struct ObjectIteratorValues
```

Comment:
After deprecating the flag there's no difference between `ObjectIteratorValues` and `IteratorValues`. `ObjectIteratorValues` should be removed.

Reply:
`ObjectIteratorValues` has `Key`, which `IteratorValues` hasn't. Do you want to add it to `IteratorValues`?

Reply:
I used inheritance (in a separate commit). If you dislike it, let me know what you do prefer, and I'll update the PR.

Reply:
> `ObjectIteratorValues` has `Key`, which `IteratorValues` hasn't

Oh, I've missed that 😞 

> I used inheritance (in a separate commit). If you dislike it, let me know what you do prefer, and I'll update the PR.

`ObjectIteratorValues` and `IteratorValues` are defined as `readonly ref struct` for performance reasons. I'd prefer having code duplication in this case.

Reply:
By the way, [the commit](https://github.com/handlebars-lang/handlebars.js/commit/3a9440f954092558275cd4c05a35ba34bcbfa210) that @yoshigev referenced has this message:
> Provide @key and @last value for all forms of iteration.

So if I read it right, maybe we can just add Key to `IteratorValues` and get rid of the other one?

Reply:
This should work as well. In that case `@Key` should return current index for non-object iterators.

Reply:
Done